src/LocalFile.js

```diff
@@ -1,85 +1,100 @@
+    const { auth, ...createRemoteArgsWithoutAuth } = this.createRemoteArgs;
 
-        sourceObject.localFiles___NODE = [];
+    try {
+      const fileNodePromises = urlsArray.map(url => {
+        const createRemoteArgs = authFlag ?
+          { url, parentNodeId, ...this.createRemoteArgs } : 
+          { url, parentNodeId, ...createRemoteArgsWithoutAuth };
 
-        await Promise.all(
-          urls.map(async url => {
-            let fileNode;
-            try {
-              fileNode = await createRemoteFileNode({
-                url,
-                parentNodeId: sourceObject.id,
-                ...this.createRemoteArgs
-              });
-            } catch (e) {
-              console.log(e);
-            }
+        return createRemoteFileNode(createRemoteArgs);
+      });
+      const fileNodes = await Promise.all(fileNodePromises);
 
-            if (fileNode) {
-              sourceObject.localFiles___NODE.push(fileNode.id);
-            }
-          })
-        );
-      })
-    );
-    return payload;
+      return this.validateFileNodes(fileNodes);
+    } catch (e) {
+        const URLStrings = urlsArray.reduce((URLString, url, i) => URLString + `URL ${i + 1}: ` + url + '\n', '');
+        console.log("\x1b[1;31m\u2715\x1b[0m We were unable to download images that stripe was pointing at\n" + URLStrings + `Error: ${e.message}\n`);
+      return null;
+    }
   }
 
-  async downloadFileNode(payload) {
-    const updatedData = payload.data.map(async file => {
-      let fileNode;
-      try {
-        fileNode = await createRemoteFileNode({
-          url: file.url,
-          parentNodeId: file.id,
-          ext: `.${file.type}`,
-          ...this.createRemoteArgs
-        });
-      } catch (e) {
-        console.log(e);
-      }
+  // File types objects have images that are hosted on Stripes servers.
+  async downloadStripeHostedFile(url, type, parentNodeId) {
+    try {
+      const fileNode = await createRemoteFileNode({
+        url,
+        ext: `.${type}`,
+        parentNodeId,
+        ...this.createRemoteArgs
+      });
 
-      if (fileNode) {
-        file.localFiles___NODE = [fileNode.id];
-      }
-      return file;
-    });
-    payload.data = await Promise.all(updatedData);
-    return payload;
+      return this.validateFileNodes(fileNode);
+    } catch (e) {
+      console.log(`\x1b[1;31m\u2715\x1b[0m We were unable to download images that stripe was hosting\nURL: ${url}\n` + `Error: ${e.message}\n`);
+      return null;
+    }
   }
 
-  // Access nested objects with path given as array.
-  getNestedObject(object, path) {
-    return path.reduce((obj, key) => obj[key], object);
+  // Accepts both a single file node and an array of file nodes
+  validateFileNodes(fileNodes) {
+    const fileNodesArray = convertToArray(fileNodes);
+    const validFileNodes = fileNodesArray.filter(node => node);
+    if (validFileNodes.length < 1) throw new Error("Failed to process remote content");
```

Comment:
Another error message nitpick: mimicking cRFN's "Failed to process remote content" might make debugging a little more confusing down the line. It would be a good idea to rephrase to something distinct.